### ISSUE TITLE
[indent] False positive when concatenating more than a function call with an object inside it

### ISSUE THREAD

**Issue by euberdeveloper** [typescript-eslint/typescript-eslint#2888]
- [ X] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
I think this is the right indentation

![image](https://user-images.githubusercontent.com/33126163/102630084-61427f80-414c-11eb-88df-bfb6ebfb2ecc.png)

but the rule wants this instead

![image](https://user-images.githubusercontent.com/33126163/102630180-833c0200-414c-11eb-9feb-7fbb553c86ce.png)

```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": ["error", 4]
  }
}
```

```TS
// your repro code case
export function validatePostOrPutLibraries(body: any): boolean {
    const schema = Joi.object({
        name: Joi
            .string()
            .min(1)
            .max(1000)
            .required()        
    })
    .required()
    .unknown();

    return schema.validate(body).error === undefined;
}
```

**Expected Result**

I expected that my code was right

**Actual Result**

As in the screenshot, the rule expects a wrong indentation, with extra spaces

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.4.0` |
| `@typescript-eslint/parser`        | `^4.4.0` |
| `TypeScript`                       | `^4.1.3` |
| `ESLint`                           | `^7.15.0` |
| `node`                             | `v14.15.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1824

Use the search next time please.
